### PR TITLE
Ensure docs can build

### PR DESCRIPTION
**Pull request by tdeebswihart**
**What changed?**
I vendored the protos necessary to generate docs _and_ added docs to the normal build process.

**Why?**
So we can ensure our docs build _before_ we merge to main. We'll need these for the gogo->google protobuf transition anyways.

**Breaking changes**

### CODE REVIEW

**Comment by tdeebswihart** [comment]
I'm not happy about placing the google protos in the root but buf won't function with them elsewhere. If we wanted to do this properly we'd need to lay things out as per https://buf.build/docs/build/usage#2.-modules-and-workspaces, which won't work with the `temporal` folder at the top level.

We can fix that if we care to

File: buf.yaml [hunk 1]
```diff
@@ -1,6 +1,4 @@
 version: v1
-deps:
-  - buf.build/googleapis/googleapis
```
Comment (stephanos) [inline buf.yaml:3]: I saw you added a comment on this PR to explain why this was vendored; could we add this somewhere around here for future reference?

**Review by stephanos — APPROVED** [review]

File: google/api/annotations.proto
Comment (stephanos) [inline google/api/annotations.proto:1]: The "googleapis" don't have a version, do they? If they do, the file path could reflect that. If they don't, then ignore my comment :)

File: buf.yaml (hunk 1 above)
Comment (tdeebswihart) [reply buf.yaml:3]: We need to remove the dep because otherwise buf will use this _and_ our vendored version and break.

File: google/api/annotations.proto
Comment (tdeebswihart) [reply google/api/annotations.proto:1]: They don't. It's irritating

File: buf.yaml (hunk 1 above)
Comment (tdeebswihart) [reply buf.yaml:3]: Since we don't use buf to build we need to vendor it ourselves

**Review by yiminc — APPROVED** [review]

File: buf.yaml (hunk 1 above)
Comment (stephanos) [reply buf.yaml:3]: Sorry, what I meant to say is: can we put a comment inside of buf.yaml or the README to explain why we _vendored_ these protos? It won't be obvious to anyone without finding your comment on this PR.

Comment (stephanos) [reply buf.yaml:3]: @tdeebswihart could we still add a little note to that effect somewhere?

Comment (stephanos) [reply buf.yaml:3]: @tdeebswihart ping

**Pull request merged**